dimod/decorators.py

```diff
@@ -497,3 +498,21 @@ def wrapper(obj, *args, **kwargs):
         return method(*args, **kwargs)
 
     return wrapper
+
+def unique_variable_labels(f):
+    """Decorator to assign unique labels to variables when no label is passed.
+
+    Designed to be applied to variable methods, :meth:`.dimod.Binary`,
+    :meth:`.dimod.Spin`, :meth:`.dimod.Integer`.
+
+    """
+    @wraps(f)
+    def conditional_unique_label(*args, **kwargs):
```

Comment:
why not `conditional_unique_label(label=None, **kwargs)` so that it more closely matches the signature we're trying to replace.